int8 last_act
---
int8 next_act
